src/renderer/main.tsx

```diff
@@ -207,18 +64,9 @@ export const Main = memo(({ url }: MainProps) => {
         )
     );
 
-    const [pythonInfo, setPythonInfo] = useState<PythonInfo>();
-    useAsyncEffect(
-        () => ({
-            supplier: () => ipcRenderer.invoke('get-python'),
-            successEffect: setPythonInfo,
-        }),
-        []
-    );
+    if (connectionState === 'failed') return null;
 
-    if (nodesQuery.isError) return null;
-
-    if (!nodesInfo || !pythonInfo || !ready) {
+    if (connectionState === 'connecting' || schemata.schemata.length === 0 || !ready) {
```

Comment:
Why `schemata.schemata.length === 0`?

Reply:
Is there a better way to check if no nodes are available? If that's not what you mean, I mentioned on discord why I did this.

Reply:
> it's setting the connectionState to 'connected' before everything is actually defined, so like it connects and goes through everything, but the schemamap is still empty

If you mean this, then I have another question: why did you not fix `connectionState` being `connected` before the data is available? This is a bug in `BackendContext` and shouldn't be fixed here with a work around.

Reply:
ah ok that makes sense